### PR TITLE
Upgradify

### PR DESCRIPTION
**Pull request by StoyanD**
adding open zepplin plugin for proxy for claims, refactoring code and tests

### CODE REVIEW

File: contracts/EcoClaim.sol [hunk 1]
```diff
@@ -176,17 +176,17 @@ contract EcoClaim is Ownable, EIP712("EcoClaim", "1") {
     /**
      * The eco ERC20 contract
      */
-    IECO public immutable _eco;
+    IECO public _eco;
```
Comment (nknavkal) [inline contracts/EcoClaim.sol:179]: why are these no longer immutable

File: test/utils/fixtures.ts [hunk 2]
```diff
@@ -76,18 +79,26 @@ export async function deployEcoClaim(
...
     root,
-    tree.getDepth()
-  )
-  await claim.deployed()
-  // @ts-ignore
-  return [eco, ecoX, ecoID, claim, balancedLeaves, tree, root]
+    tree.getDepth()], {
+    initializer: "initialize",
+  })
+  // const claim = await EcoClaim.deploy(
```
Comment (nknavkal) [inline test/utils/fixtures.ts:91]: remove commented code

Comment (StoyanD) [reply test/utils/fixtures.ts:91]: removed in later pr

File: contracts/EcoClaim.sol (hunk 1 above)
Comment (StoyanD) [reply contracts/EcoClaim.sol:179]: an immutable must be initialized in a constructor and the proxy pattern requires that we dont use constructors for initialization... so immutables have to go

**Review by nknavkal — APPROVED** [review]

**Pull request merged**